File: js/source.js
function sieve(n) {
    const prime = Array(n+1).fill(1);
    var p = 2;

    while(p * p <= n) {
        if(prime[p]) {
            for(i = p * p; i < n + 1; i += p) {
                prime[i] = 0;
            }
        }
        p++;
    }

    return prime;
}

console.log("--- Hello from JavaScript");
const num = 100,
prime     = sieve(num);

for(i = 2; i < num; i++) {
    if(prime[i]) console.log(i);
}

File: python/source.py
import numpy as np


def sieve(n):
    prime = np.ones((n + 1), dtype=int)
    p = 2
    while p * p <= n:
        if prime[p] == True:
            for i in range(p * p, n + 1, p):
                prime[i] = False
        p += 1

    return prime


if __name__ == "__main__":
    print("--- Hello from Python")
    num = 100
    prime = sieve(num)

    for p in range(2, num + 1):
        if prime[p]:
            print(p)
